src/domain/Lists/DraggableList/DraggableList.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+import {
+  DragDropContext,
+  Draggable,
+  DropResult,
+  Droppable
+} from 'react-beautiful-dnd'
+import styled, { css } from 'styled-components'
+
+import { Props, Variables } from '../../../common'
+
+interface IDraggableListProps<T> {
+  /** The array of items */
+  items: T[]
+  /** Function to update the array of items when they have been re-arranged */
+  setItems: (nodes: T[]) => void
+  /** An array of ReactNode that should represent the array of items */
+  children: React.ReactNode[]
+  /** The data-component-context */
+  componentContext?: string
+}
+
+interface IListItemProps {
+  isDragging: boolean
+  children: React.ReactNode
+}
+
+const StyledListItem = styled.div<IListItemProps>`
+  border: 1px solid ${Variables.Color.n250};
+  background-color: ${Variables.Color.n100};
+  padding: ${Variables.Spacing.sMedium}px;
+  border-radius: ${Variables.Style.borderRadius}px;
+  box-shadow: ${Variables.BoxShadow.bsLv1Static};
+
+  ${(props) => props.isDragging && css`
+    box-shadow: ${Variables.BoxShadow.bsLv3Active};
+    background-color: ${Variables.Color.n150};
+    cursor: grabbing !important;
+  `}
+`
+
+const StyledListArea = styled.div`
+  display: grid;
+  row-gap: ${Variables.Spacing.sXSmall}px;
+`
+
+function DraggableList<T> ({ items, setItems, children, componentContext }: IDraggableListProps<T>) {
+  const onDragEnd = (result: DropResult) => {
+    if (!result.destination) {
+      return
+    }
+    const { source, destination } = result
+
+    const newItems = [...items]
+    const [removed] = newItems.splice(source.index, 1)
+    newItems.splice(destination.index, 0, removed)
+    setItems(newItems)
+  }
+
+  return (
+    <span
+      data-component-type={Props.ComponentType.DraggableList}
+      data-component-context={componentContext}
+    >
+      <DragDropContext onDragEnd={onDragEnd}>
+        <Droppable droppableId='droppable'>
```

Comment:
Should we pass in id for the droppable? It may not work with two droppable with same id in a page.

Reply:
I'm changing it to set to a random UUID, unless you think we need to be able to set it specifically?

Reply:
Just saving the user entering another prop if it's not really something they need to know